SPEC.md

```diff
@@ -128,15 +131,29 @@ Examples: `1.2.3`, `>= 1.0.0` or `~> 2.0`.
 
 A local path (String).
 
-The library will be linked from the local path. The `version` attribute
-isn't required but will be used if present to validate the dependency.
+The library will be linked from the local path. The `version` attribute is not
+required but will be used if present to validate the dependency.
 
 #### git
 
 A Git repository URL (String).
 
+The URL may be of [any protocol supported by the git command line client](https://git-scm.com/docs/git-clone#_git_urls_a_id_urls_a),
+which may include:
+
+- `ssh://`
+- `git://`
+- `https://`
+- `http://`
+- `ftps://`
+- `ftp://`
+- implied ssh protocol (e.g., `user@example.com:subdir/repo.git`)
+
+Just the same as the local git command line client, `ssh_config` and associated
+`netrc` settings will be honored for authentication-protected repositories.
```

Comment:
Without this I'm concerned how others might guess at specifying different credentials depending on the source.

Would there be a more appropriate place to include this idiosyncrasy?

For all protocols (except ssh) we seem to assume it will be wide open to anonymous users. In reality, as is the case with `Bundler`, we integrate the native configuration files for each protocol. I would imagine forgoing these configuration files in `libgit2` would break far more functionality than it would help. Thoughts?